projects/dotcms-field-elements/src/components/dot-form/dot-form.tsx

```diff
@@ -0,0 +1,60 @@
+import { Component, Element, Event, EventEmitter, Listen, Prop } from '@stencil/core';
+
+@Component({
+    tag: 'dot-form'
+})
+export class DotFormComponent {
+    @Element() el: HTMLElement;
+    @Event() formSubmit: EventEmitter;
+    @Prop() submitLabel: string;
+    @Prop() resetLabel: string;
+
+    @Prop({ mutable: true }) _formValues = {};
+
+    @Listen('valueChanges')
+    onValueChanges(event: CustomEvent): void {
+        this._formValues[event.detail.name] = event.detail.value;
+    }
+
+    @Listen('stateChanges')
+    onStateChanges(event: CustomEvent): void {
+        // refresh variables from hostData
+        console.log('----onStateChanges', event);
+    }
+
+    hostData() {
+        // TODO: do validation here
+        return {
+          'class': { 'is-open': this._formValues },
+          'aria-hidden': this._formValues ? 'false' : 'true'
+        };
+      }
+
+    handleSubmit(evt: Event): void {
+        evt.preventDefault();
+        this.formSubmit.emit(this._formValues);
+    }
+
+    resetForm(): void {
+        this._formValues = {};
+        const elementsCount = this.el.children[0].children.length;
+        for (let i = 0, total = elementsCount; i < total - 2; i++) {
+            try {
+                const element: any = this.el.children[0].children[i];
+                element.reset();
+            } catch (error) {
+                console.log('Error:', error);
+            }
+        }
+    }
+
+    render() {
+        return (
+            <form onSubmit={(evt: Event) => this.handleSubmit(evt)}>
+                <slot />
```

Comment:
maybe this component should receive the fields and do this

https://github.com/dotCMS/core-web/pull/978/files#diff-6879e278aa3da8a9cce8a45acf08adecR52

I think DotApiForm should get and sent data, for example listen the submit event and send the request (through dotApiContentType), and dot-form should do all the DOM stuff